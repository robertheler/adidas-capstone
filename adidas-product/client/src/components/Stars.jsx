import React, { Component } from 'react';
import styled from 'styled-components';

const RenderSpan = styled.span`
  font-weight: 700;
  font-style: bold;
  font-family: AdihausDIN,Helvetica;
  font-size: 11px;
  margin-right: 5px;
`;

class Stars extends Component {
  constructor(props){
    super(props);
  }

  //Render 5 stars filled appropriately
  render() {
    if (this.props.review > 4.5) {
      return (<RenderSpan>★ ★ ★ ★ ★</RenderSpan>);
    } else if (this.props.review > 3.5) {
      return (<RenderSpan>★ ★ ★ ★ ☆</RenderSpan>);
    } else if (this.props.review > 2.5) {
      return (<RenderSpan>★ ★ ★ ☆ ☆</RenderSpan>);
    } else if (this.props.review > 1.5) {
      return (<RenderSpan>★ ★ ☆ ☆ ☆</RenderSpan>);
    } else if (this.props.review > 0.5) {
      return (<RenderSpan>★ ☆ ☆ ☆ ☆</RenderSpan>);
    } else return (<RenderSpan>☆ ☆ ☆ ☆ ☆</RenderSpan>);
  }
}

export default Stars;