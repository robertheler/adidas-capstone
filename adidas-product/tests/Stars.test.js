import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Stars from '../client/src/components/Stars';

// json data will be used to generate props
var data = require('../db/data.json');

function setup() {
  const props = {
    color: 0,
    product: data[0]
  };
  const wrapper = shallow(<Stars />);
  return { wrapper, props };
}

describe('Title Component Test Suite', () => {
  const { wrapper, props } = setup();

  it('It should render', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('It should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});