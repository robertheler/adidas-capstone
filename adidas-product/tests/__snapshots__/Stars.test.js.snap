// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component Test Suite It should render correctly 1`] = `
<styled.span>
  ☆ ☆ ☆ ☆ ☆
</styled.span>
`;
